clients/python/tests/test_list_inferences.py

```diff
@@ -572,10 +572,10 @@ async def test_simple_query_chat_function_async(
         output_0 = output[0]
         assert output_0.type == "text"
         # Type narrowing: we know it's a Text block
-        assert isinstance(output_0, Text)
+        assert isinstance(output_0, ContentBlockChatOutputText)
```

Comment:
do the types here need to be trees? Can we have DAGs instead?

Reply:
i think technically this is expressing that `ContentBlockChatOutput::Text` is a different "text" type from `StoredInputMessageContent::Text` even though their internal values are both `Text` and they both have a `type` field, so this seems ~correct to me; we could instead have a shared `TaggedText` class that's `{ type: str = "text" } & Text` that might be cleaner today from the developer's perspective, but if things ever change then it becomes annoying to undo?